### ISSUE TITLE
Transformer Architecture and other issues

### ISSUE THREAD

**Issue by da03** [salesforce/nonauto-nmt#3]
In this implementation transformer decoder layer i is attending to encoder layer i output, which seems to be different from Google's original transformer implementation which always attends to the last layer output.

https://github.com/salesforce/nonauto-nmt/blob/efcbe4f2329b140ac3ce06abb6409457cebc8e49/model.py#L601

Plus, the provided scripts seem to contain options not supported by run.py, and many train_fast.sh relies on -load_from old_fast_model, which further makes the paper's results hard to reproduce.

**Comment by jekbradbury** [comment]
You’re right, and this is likely one of the reasons our autoregressive reproduction results were slightly off from the original paper (though I didn’t notice until several months later). I believe that paper’s description was a little ambiguous on this point, but the code is quite clear that it attends to the last layer.
